classes/class-p4-search.php

```diff
@@ -43,45 +45,195 @@ public function __construct( $search_query, $templates = [ 'search.twig', 'archi
 				$this->context = $context;
 			} else {
 				$this->context = Timber::get_context();
+				$this->handle_submit( $this->context );
 
+				if ( $this->search_query ) {
+					$this->all_posts    = $this->get_posts();
+					$this->current_page = get_query_var( 'paged' ) ? get_query_var( 'paged' ) : 1;
+					$this->posts        = array_slice( $this->all_posts, ( $this->current_page - 1 ) * self::POSTS_PER_PAGE, self::POSTS_PER_PAGE );
+				}
+				$this->set_context( $this->context );
+			}
+		}
+
+		/**
+		 * Handle form submit.
+		 *
+		 * @param array $context Associative array with the data to be passed to the view.
+		 */
+		public function handle_submit( &$context ) {
+			if ( 'GET' === $_SERVER['REQUEST_METHOD'] ) {
+
+				// Handle submitted sort options.
+				$selected_sort = filter_input( INPUT_GET, 'orderby', FILTER_SANITIZE_STRING );
+				if ( ! in_array( $selected_sort, array_keys( $context['sort_options'] ), true ) ) {
+					$context['selected_sort'] = self::DEFAULT_SORT;
+				} else {
+					$context['selected_sort'] = $selected_sort;
+				}
+
+				// Handle submitted filter options.
+				if ( is_array( $_GET['f'] ) ) {
+					foreach ( $_GET['f'] as $type => $filter_type ) {
+						foreach ( $filter_type as $name => $id ) {
+
+							$id = filter_var( $id, FILTER_VALIDATE_INT );
+							if ( $this->validate( $id ) ) {
+								$this->filters[ $type ][] = [
+									'id'   => $id,
+									'name' => $name,
+								];
+							}
+						}
+					}
+				}
+			}
+		}
+
+		/**
+		 * Applies user selected filters to the search if there are any and gets the filtered posts.
+		 * If there are not then uses Timber's get_posts to retrieve all of them (up to the limit set).
+		 *
+		 * @return array The posts of the search.
+		 */
+		protected function get_posts() : array {
+			global $wp_query;
+
+			if ( ! $wp_query->is_main_query() || ! $wp_query->is_search() ) {
```

Comment:
I don't understand the purpose for these checks, but maybe it is better to move them in a an upper level (search.php).
I think it would be better for the separation of concerns.